src/main/java/com/nirima/jenkins/plugins/docker/DockerContainerWatchdog.java

```diff
@@ -121,18 +137,23 @@ protected void execute(TaskListener listener) throws IOException, InterruptedExc
         Instant snapshotInstance = this.clock.instant();
         Map<String, Node> nodeMap = loadNodeMap();
         
-        for (Cloud c : getAllClouds()) {
-            if (!(c instanceof DockerCloud)) {
-                continue;
+        try {
+            for (Cloud c : getAllClouds()) {
+                if (!(c instanceof DockerCloud)) {
+                    continue;
+                }
+                DockerCloud dc = (DockerCloud) c;
+                LOGGER.info("Checking Docker Cloud '{}'", dc.getDisplayName());
+                listener.getLogger().println(String.format("Checking Docker Cloud %s", dc.getDisplayName()));
+                
+                csmMerged = processCloud(dc, nodeMap, csmMerged, snapshotInstance);
             }
-            DockerCloud dc = (DockerCloud) c;
-            LOGGER.info("Checking Docker Cloud '{}'", dc.getDisplayName());
-            listener.getLogger().println(String.format("Checking Docker Cloud %s", dc.getDisplayName()));
-            
-            csmMerged = processCloud(dc, nodeMap, csmMerged, snapshotInstance);
+    
+            cleanUpSuperfluousComputer(nodeMap, csmMerged, snapshotInstance);
+        } catch (WatchdogProcessingTimeout timeout) {
+            LOGGER.warn("Processing of cleanup watchdog took too long");
```

Comment:
In "real world" usage, I've found that it's very useful, when logging a timeout, to log the following information:
(1) How long the timeout was, specifying both the number _and_ units of the time, as the person reading the log may not know if that's milliseconds, seconds, minutes etc.
(2) When the timeout's timer started, so that whoever is looking at the log can verify that there's an entry saying "things are starting" at that time and that, exactly <timeout duration> later the timeout happens.
(3) Where the code was when the exception happened, i.e. the stacktrace.

I've spent far too many hours of my life trying to debug systems that have hung/locked-up etc and this is the information that I'd want to see in this situation...